gtsfm/averaging/translation/averaging_1dsfm.py

```diff
@@ -10,33 +10,46 @@
 
 Authors: Jing Wu, Ayush Baid, Akshay Krishnan
 """
+import random
+from collections import defaultdict
+from enum import Enum
 from typing import Dict, List, Optional, Tuple
 
 import gtsam
 import numpy as np
 from gtsam import MFAS, BinaryMeasurementsUnit3, BinaryMeasurementUnit3, Point3, Pose3, Rot3, TranslationRecovery, Unit3
+from scipy import stats
 
 import gtsfm.utils.geometry_comparisons as comp_utils
 import gtsfm.utils.metrics as metrics_utils
 import gtsfm.utils.coordinate_conversions as conversion_utils
+import gtsfm.utils.logger as logger_utils
 from gtsfm.averaging.translation.translation_averaging_base import TranslationAveragingBase
 from gtsfm.evaluation.metrics import GtsfmMetric, GtsfmMetricsGroup
 
 # Hyperparameters for 1D-SFM
 # maximum number of times 1dsfm will project the Unit3's to a 1d subspace for outlier rejection
 MAX_PROJECTION_DIRECTIONS = 200
 OUTLIER_WEIGHT_THRESHOLD = 0.1
+MAX_KDE_SAMPLES = 2000
 
 NOISE_MODEL_DIMENSION = 3  # chordal distances on Unit3
 NOISE_MODEL_SIGMA = 0.01
 HUBER_LOSS_K = 1.345  # default value from GTSAM
 
 MAX_INLIER_MEASUREMENT_ERROR_DEG = 5.0
 
+logger = logger_utils.get_logger()
 
 class TranslationAveraging1DSFM(TranslationAveragingBase):
     """1D-SFM translation averaging with outlier rejection."""
 
+    class ProjectionSamplingMethod(Enum):
```

Comment:
very nice. I like it. would a string enum value be a little clearer? maybe not. or is `auto()` simpler?
```python
class ProjectionSamplingMethod(str, Enum):
     SAMPLE_INPUT_MEASUREMENTS: str = "SAMPLE_INPUT_MEASUREMENTS"
```

Reply:
I think auto is better. A string is probably inefficient, I dont see a use for it.